test/util/util.go

```diff
@@ -680,12 +662,22 @@ func ExpectPodsFinalized(ctx context.Context, k8sClient client.Client, keys ...t
 	for _, key := range keys {
 		createdPod := &corev1.Pod{}
 		gomega.EventuallyWithOffset(1, func(g gomega.Gomega) {
-			g.Expect(k8sClient.Get(ctx, key, createdPod)).To(gomega.Succeed())
+			g.Expect(client.IgnoreNotFound(k8sClient.Get(ctx, key, createdPod))).To(gomega.Succeed())
 			g.Expect(createdPod.Finalizers).Should(gomega.BeEmpty(), "Expected pod to be finalized")
 		}, Timeout, Interval).Should(gomega.Succeed())
 	}
 }
 
+func ExpectWorkloadsFinalized(ctx context.Context, k8sClient client.Client, keys ...types.NamespacedName) {
```

Comment:
As in the other comment, I would suggest name like `ExpectWorkloadsGoneOrFinalized` to better express the intention. In some cases we may want to assert only that there are no finalizers, but the workload still exists (because no-one send the delete request)?

Reply:
Renamed both functions.